### PR TITLE
Plugin

### PR DESCRIPTION
**Pull request by StefMa**

### CODE REVIEW

File: gitversioner/build.gradle [hunk 1]
```diff
@@ -0,0 +1,43 @@
+apply plugin: 'kotlin'
+apply plugin: 'com.novoda.bintray-release'
+
+group = 'com.pascalwelsch.gitversioner'
+version = '0.0.1-SNAPSHOT'
+
+buildscript {
+    ext.kotlin_version = '1.1.2-4'
+    configure([repositories, project.repositories]) {
+        maven { url 'https://repo.gradle.org/gradle/repo' }
```
Comment (StefMa) [inline gitversioner/build.gradle:10]: As the `kotlin-gradle-plugin` is available via the gradle plugin portal you can have a much much easier implementation of the plugin:
```groovy
plugins {
  id "org.jetbrains.kotlin.jvm" version "1.1.2-2"
}
```
See https://plugins.gradle.org/plugin/org.jetbrains.kotlin.jvm

File: gitversioner/build.gradle [hunk 2]
```diff
@@ -0,0 +1,43 @@
+apply plugin: 'kotlin'
+apply plugin: 'com.novoda.bintray-release'
+
+group = 'com.pascalwelsch.gitversioner'
+version = '0.0.1-SNAPSHOT'
+
+buildscript {
+    ext.kotlin_version = '1.1.2-4'
```
Comment (StefMa) [inline gitversioner/build.gradle]: Why declare it inside a buildscript and add it to the `ExtensionContainer`? 🤔 
I think it is easier to read to put it out of the buildscript (directly under `version`) and use it like it is :)

File: gitversioner/build.gradle [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+sourceSets {
+    main.java.srcDirs += 'src/main/kotlin'
+}
+
+publish {
+    userOrg = 'passsy'
+    groupId = 'com.pascalwelsch.gitversioner'
+    artifactId = 'gitversioner'
+    publishVersion = '0.1.0'
```
Comment (StefMa) [inline gitversioner/build.gradle]: Use `version` you have declared above

File: gitversioner/build.gradle [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+    testCompile "com.nhaarman:mockito-kotlin:1.5.0"
+}
+
+sourceSets {
+    main.java.srcDirs += 'src/main/kotlin'
+}
+
+publish {
+    userOrg = 'passsy'
+    groupId = 'com.pascalwelsch.gitversioner'
```
Comment (StefMa) [inline gitversioner/build.gradle]: Use the `groupId` you have declared above

File: gitversioner/build.gradle (hunk 2 above)
Comment (passsy) [reply gitversioner/build.gradle]: code inside `buildscript` will be executed before code outside of `buildscript` which is required to get the kotlin plugin dependency. Also this is how kotlin is added everywhere, I'll stick to the standard

File: gitversioner/build.gradle (hunk 1 above)
Comment (passsy) [reply gitversioner/build.gradle:10]: the gradle maven repo contains the latest versions. `1.1.2-2` has problems with latest stable Android Studio, `1.1.2-4` is required.

**Comment by passsy** [comment]
Moved to a new repository. Could break builds of users of the current script if it isn't on master anymore

https://github.com/passsy/gradle-gitVersioner-plugin

**Comment by StefMa** [comment]
Why not just use another branch and make them as default? 🤔

**Comment by passsy** [comment]
I want `master` to be the default branch

**Comment by StefMa** [comment]
Lol okay 👌 😂🤣

**Pull request closed without merging**